contents/handbook/engineering/clickhouse/clusters.mdx

```diff
@@ -0,0 +1,220 @@
+
+### Tiered storage
+
+One of the nice features of our data is that recent data is the hottest both for reads and writes. This is a perfect fit for tiered storage.
+We can basically read and write to local ephemeral NVMe (with solid backup strategies) and then move the data to cheaper EBS volumes as it ages out.
+We currently do this using tiered storage configured simply by setting the `storage` configs in ClickHouse, but we eventually will want to move to setting TTLs on tables and having ClickHouse manage the tiering for us consistently.
+
+https://altinity.com/blog/2019-11-29-amplifying-clickhouse-capacity-with-multi-volume-storage-part-2
+
+### Monitoring
+
+https://grafana.prod-us.posthog.dev/d/vm-clickhouse-cluster-overview/clickhouse-cluster-overview?from=now-3h&to=now&timezone=utc&refresh=30s
+
+
+## Prod-EU
+
+Prod EU is our main production cluster for the EU.
+
+It is made up of the following setup:
+
+- 8 shards
+- 2 replicas
+- `m6g.8xlarge`
+- 32 vCPUs
+- Graviton2 CPU
+- 128 GiB RAM
+- Max Network 12 Gbps
+- Max EBS Throughput 1187.5 MB/s
+- Single 10TB GP3 EBS volume 
+
+
+We hit a problem with having smaller shards on EU that had a significant performance impact. We were running out of memory for larger queries which was also impacting our page cache hit rate.
+This was mainly due to limiting query size restrictions to protect other users of the cluster.
+We had two solutions for this. Increase the size of the nodes...meaning double the size for each instance x 16, very expensive. Or, we could setup a coordinator node.
+The coordinator node is a topology that allows us to effectively split the storage and compute tiers of the cluster into two pools of resources.
+We treat the current cluster of small nodes with many shards as the storage tier, they effectively are the mappers of the cluster and quickly fetch the data that we want for queries.
+We then send that relatively small data back to the coordinator and do the heavy lifting there which includes joins and aggregates.
+
+For the EU the coordinator node is:
+
+- 1 instance
+- `c7g.metal`
+- 64 vCPUs
+- metal Graviton3 CPU
+- 128 GiB RAM
+- Max Network 30 Gbps
+- Max EBS Throughput 2500 MB/s
+- 4 x 2.5TB GP3 EBS volumes
+    - JBOD configuration
+
+
+#### Coordinator schema
+
+The coordinator has distributed tables (`events`, `session_replay`) tables that point to the EU Prod cluster
+
+All non-sharded tables are replicated so that they are local to the coordinator.
+
+
+#### Coordinator future
+
+We should probably consider moving this to a `m7g.metal` if we hit any memory constraints with this, but so far we have not because of the dedicated nature of this node.
+
+We will also want to create new coodinators for multi-tennant workloads in the future. This will allow us to scale up and down easily over time, and even potentially throughout the day as the workload rises and falls.
```

Comment:
```suggestion
We will also want to create new coodinators for multi-tenant workloads in the future. This will allow us to scale up and down easily over time, and even potentially throughout the day as the workload rises and falls.
```